vrf-ad-kc/src/lib.rs

```diff
@@ -69,19 +68,20 @@ pub const fn small_cofactor_projective<C: ProjectiveCurve>() -> bool {
 /// If false then perform full subgroups checks in deserialization
 /// If true then only perform on-curve checks in deserialization,
 /// but invoke `mul_by_cofactor` in hashing and equality checks. 
-pub const fn small_cofactor<C: AffineCurve>() -> bool {
-    small_cofactor_projective::<<C as AffineCurve>::Projective>()
+pub const fn small_cofactor<C: AffineRepr>() -> bool {
+    small_cofactor_projective::<<C as AffineRepr>::Group>()
 }
 
-pub fn eq_mod_small_cofactor_projective<C: ProjectiveCurve>(lhs: &C, rhs: &C) -> bool {
+pub fn eq_mod_small_cofactor_projective<C: CurveGroup>(lhs: &C, rhs: &C) -> bool {
     if crate::small_cofactor_projective::<C>() {
-        lhs.mul(<C as ProjectiveCurve>::COFACTOR) == rhs.mul(<C as ProjectiveCurve>::COFACTOR)
+        lhs.mul_bigint(<<C as CurveGroup>::Config as CurveConfig>::COFACTOR)
+         == rhs.mul_bigint(<<C as CurveGroup>::Config as CurveConfig>::COFACTOR)
     } else { lhs == rhs }
 }
 
-pub fn eq_mod_small_cofactor_affine<C: AffineCurve>(lhs: &C, rhs: &C) -> bool {
+pub fn eq_mod_small_cofactor_affine<C: AffineRepr>(lhs: &C, rhs: &C) -> bool {
     if crate::small_cofactor::<C>() {
-        lhs.mul_by_cofactor() == rhs.mul_by_cofactor()
+        lhs.clear_cofactor() == rhs.clear_cofactor() // mul_by_cofactor is fine here though
```

Comment:
you should probably leave it as `clear_cofactor`, if you just care about it being in the right prime order subgroup, as it might have faster implementations (currently it does for bls12-377 & bls12-381, but might get implemented for other curves in the future).
If you care about it being the result of multiplying by cofactor proper, you should use `mul_by_cofactor`

Reply:
this code only runs if the cofactor is 8 or less, but yeah maye someone would increase that.